src/frontend/packages/lib_common/src/queries/fetchList.tsx

```diff
@@ -39,5 +40,5 @@ export const fetchList: QueryFunction<any, FetchListQueryKey> = async ({
     throw new Error(`Failed to get list of ${name}.`);
   }
 
-  return await response.json();
+  return (await response.json()) as Promise<T>;
```

Comment:
```suggestion
  return (await response.json()) as T;
```

Reply:
or really return the promise ?

Reply:
No, it's an async function, it already return a Promise and in fact the value of an awaited response.json() is the expected return type not a Promise

Reply:
At the base, I stick to the legacy function definition: 
`import { MutationFunction } from 'react-query';`
`type MutationFunction<TData = unknown, TVariables = unknown> = (variables: TVariables) => Promise<TData>`

So you are normally waiting a Promise, after I agree it doesn't make sense to cast to a Promise.
Should it not be that at the end? 
`return response.json() as Promise<T>;`

Reply:
If you `return response.json()` yes, you are returning a Promise.
If you `return await response.json()` you are no longer """""returning""""" a promise but you are returning the value of the json content, but since this the return of an async function, on the top level this is a Promise.
so `return await response.json() as T` and `return response.json() as Promise<T>` are both correct and pretty much similar  in this case